### PR TITLE
chore: use ruff

### PR DESCRIPTION
**Pull request by henryiii**
Trying this out. Very impressed so far. Some minor issues, but very complete, fast, and simple. No more `.flake8` file!

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -61,7 +61,6 @@ def docs(session: nox.Session) -> None:
 
     if session.posargs:
         if "--serve" in session.posargs:
-            print("Launching docs at http://localhost:8000/ - use Ctrl-C to quit")
```
Comment (henryiii) [inline noxfile.py:64]: Oh, wow, auto fixing no print statements might be something to turn off. :)

File: pyproject.toml [hunk 2]
```diff
@@ -94,14 +94,12 @@ cmake_source_dir = "scikit_build_core.setuptools.extension:cmake_source_dir"
 build.exclude = ["extern"]
 version.source = "vcs"
 build.hooks.vcs.version-file = "src/scikit_build_core/_version.py"
-
-[tool.hatch.build.force-include]
```
Comment (henryiii) [inline pyproject.toml:98]: This is a bug in the TOML crate for Rust, see https://github.com/charliermarsh/ruff/issues/1894.

File: noxfile.py (hunk 1 above)
Comment (agoose77) [reply noxfile.py]: Huh that seems a bit aggressive!

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -56,39 +52,11 @@ repos:
...
-          - flake8-print
-
-  - repo: https://github.com/pycqa/flake8
-    rev: 6.0.0
-    hooks:
-      - id: flake8
-        exclude: docs/conf.py
-        additional_dependencies: *flake8_dependencies
+      - id: ruff
+        args: ["--fix"]
```
Comment (henryiii) [inline .pre-commit-config.yaml:59]: This really should have --force-exclude, but the next version of the Action won't need it. https://github.com/charliermarsh/ruff-pre-commit/pull/20

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -56,39 +52,11 @@ repos:
       - id: blacken-docs
         additional_dependencies: [black==22.8.0]
 
-  - repo: https://github.com/PyCQA/isort
-    rev: 5.11.4
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: v0.0.226
```
Comment (henryiii) [inline .pre-commit-config.yaml:56]:
```suggestion
    rev: v0.0.228
```

File: .pre-commit-config.yaml (hunk 3 above)
Comment (henryiii) [reply .pre-commit-config.yaml:59]: And it's out!

**Comment by henryiii** [comment]
pre-commit.ci autofix

**Pull request merged**